tiny86/src/Syscall/Read.hs

```diff
@@ -0,0 +1,12 @@
+module Syscall.Read where
+
+import Syscall.Internal
+import Clash.Prelude
+
+-- NOTE(jl): inline translation of MSP430 DFA, see: sieve:circuit/msp430/regfile.v:204-207.
+syscallReadDFA :: SyscallDFA
```

Comment:
I think I understand this after comparing against Receive.hs, but would you be able to add some comments explaining each case?